includes/multi-currency/FrontendCurrencies.php

```diff
@@ -215,6 +216,16 @@ public function init_order_currency( $order ) {
 		return $order->get_id();
 	}
 
+	/**
+	 * Gets the order id from the wp query_vars and then calls init_order_currency.
+	 *
+	 * @return void
+	 */
+	public function init_order_currency_from_query_vars() {
+		global $wp;
+		$this->init_order_currency( $wp->query_vars['order-pay'] );
```

Comment:
I would add a check if the `query_vars` has an `order-pay` key first here before sending it to the function. What do you think?

Reply:
I think it's a good idea, and updated in https://github.com/Automattic/woocommerce-payments/pull/3323/commits/1d703be4bbd253c28b5ad09735410b4f457a2363 . 

The reason I didn't do it initially is due to WooCommerce core _shouldn't_ call the `before_woocommerce_pay` action if the `order-pay` key is empty: https://github.com/woocommerce/woocommerce/blob/5.9.0/includes/shortcodes/class-wc-shortcode-checkout.php#L57-L80 . It's better to double check, though, thanks for spotting this.

Reply:
Thanks for doing this :) IMO, It's good to be extra defensive in those cases.